source/common/redis/command_splitter_impl.cc

```diff
@@ -21,43 +21,50 @@ RespValuePtr Utility::makeError(const std::string& error) {
   return response;
 }
 
-SplitRequestPtr AllParamsToOneServerCommandHandler::startRequest(const RespValue& request,
-                                                                 SplitCallbacks& callbacks) {
-  std::unique_ptr<SplitRequestImpl> request_handle(new SplitRequestImpl(callbacks));
-  request_handle->handle_ =
-      conn_pool_.makeRequest(request.asArray()[1].asString(), request, *request_handle);
-  if (!request_handle->handle_) {
-    callbacks.onResponse(Utility::makeError("no upstream host"));
+SplitRequestPtr SimpleRequest::create(ConnPool::Instance& conn_pool,
+                                      const RespValue& incoming_request,
+                                      SplitCallbacks& callbacks) {
+  std::unique_ptr<SimpleRequest> request_ptr =
+      std::unique_ptr<SimpleRequest>{new SimpleRequest(callbacks)};
```

Comment:
Why not `std::unique_ptr<SimpleRequest> request_ptr{new SimpleRequest(callbacks)};` ?